### PR TITLE
Core/Battlegrounds: Possible fixed unbalanced battlegrounds (e.g. WSG…

### LINKED ISSUE #49: 10+ players in warsong per faction [$15]
Originally reported https://github.com/chromiecraft/chromiecraft/issues/335

#### WHAT CLIENT DO YOU PLAY ON?

- [ ] enGB
- [X] enUS
- [ ] other (specify)

##### FACTION

- [ ] Alliance
- [ ] Horde

##### CONTENT PHASE:

- [ ] Generic
- [ ] 1-19
- [X] 20-29
- [ ] 30-39

##### CURRENT BEHAVIOUR:

10+ players in warsong.

##### EXPECTED BLIZZLIKE BEHAVIOUR:

Max players per faction is 10.

##### EXTRA NOTES:

Not sure how this happened.
Happen for both faction.

Capture d’écran (13)
Capture d’écran (14)

##### AC HASH/COMMIT:

https://github.com/chromiecraft/azerothcore-wotlk/commit/9d4eebc61fec3b9701ae94292e8d40962804b676

##### OPERATING SYSTEM:

Ubuntu 20.04

##### MODULES:

- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [lua-LevelUpReward](https://github.com/55Honey/Acore_LevelUpReward)

##### OTHER CUSTOMIZATIONS:

None.

##### SERVER:

ChromieCraft

<bountysource-plugin>

---
There is a **[$15 open bounty](https://www.bountysource.com/issues/98051715-10-players-in-warsong-per-faction?utm_campaign=plugin&utm_content=tracker%2F146727645&utm_medium=issues&utm_source=github)** on this issue. Add to the bounty at [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F146727645&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #335: 10+ players in warsong per faction
#### WHAT CLIENT DO YOU PLAY ON?

- [ ] enGB
- [X] enUS
- [ ] other (specify)

##### FACTION

- [ ] Alliance
- [ ] Horde

##### CONTENT PHASE:

- [ ] Generic
- [ ] 1-19
- [X] 20-29
- [ ] 30-39

##### CURRENT BEHAVIOUR:

10+ players in warsong.

##### EXPECTED BLIZZLIKE BEHAVIOUR:

Max players per faction is 10.

##### EXTRA NOTES:

Not sure how this happened.
Happen for both faction.

Capture d’écran (13)
Capture d’écran (14)

##### AC HASH/COMMIT:

https://github.com/chromiecraft/azerothcore-wotlk/commit/9d4eebc61fec3b9701ae94292e8d40962804b676

##### OPERATING SYSTEM:

Ubuntu 20.04

##### MODULES:

- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [lua-LevelUpReward](https://github.com/55Honey/Acore_LevelUpReward)

##### OTHER CUSTOMIZATIONS:

None.

##### SERVER:

ChromieCraft

### LINKED ISSUE #2316: More than 10 ppl on one team in WSG... again
### What client do you play on?

enUS

### Faction

- [X] Alliance
- [ ] Horde

### Content Phase:

- [X] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59

### Current Behaviour

In chat pane, you can see healers confirming via their raid panes. You can also confirm by mousing over the zone map, although that would require a video to submit as proof.

WoWScrnShot_111321_060025

### Expected Blizzlike Behaviour

10 max per team in WSG 

### Source

common knowledge of wow mechanics

### Steps to reproduce the problem

Do whatever you guys did the first time you fixed this, ngl really feels like our efforts cheapened when this can just crop right back up

### Extra Notes

_No response_

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/9e3332dc208215f60d62b6eef3c9e804464708da

### Operating system

Windows 10

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)
- [lua-zone-debuff](https://github.com/55Honey/Acore_ZoneDebuff)

### Customizations

None

### Server

ChromieCraft

### PR DESCRIPTION
**Pull request by ghost**
… 11v9).

Fixed azerothcore/mod-cfbg#49
Fixed chromiecraft/chromiecraft#335
Fixed chromiecraft/chromiecraft#2316

### CODE REVIEW

**Comment by ghost** [comment]
@FrancescoBorzi As it's very hard to test, maybe just merge it?

**Comment by FrancescoBorzi** [comment]
> @FrancescoBorzi As it's very hard to test, maybe just merge it?

@UltraNix if you have at least tested that this does not cause any regression (i.e. check that the normal flow still works fine), sure we can merge it.

File: src/CFBG.cpp [hunk 1]
```diff
@@ -329,8 +329,15 @@ bool CFBG::IsAvgIlvlTeamsInBgEqual(Battleground* bg)
 void CFBG::ValidatePlayerForBG(Battleground* bg, Player* player, TeamId teamId)
 {
     BGData bgdata = player->GetBGData();
-    bgdata.bgTeamId = teamId;
-    player->SetBGData(bgdata);
+    if (bgdata.bgTeamId != teamId)
+    {
+        bg->DecreaseInvitedCount(bgdata.bgTeamId);
```
Comment (Winfidonarleyan) [inline src/CFBG.cpp:334]: no need

File: src/CFBG.cpp [hunk 2]
```diff
@@ -329,8 +329,15 @@ bool CFBG::IsAvgIlvlTeamsInBgEqual(Battleground* bg)
...
+
+        bgdata.bgTeamId = teamId;
+        player->SetBGData(bgdata);
+
+        bg->IncreaseInvitedCount(bgdata.bgTeamId);
```
Comment (Winfidonarleyan) [inline src/CFBG.cpp:339]: no need

File: src/CFBG.cpp (hunk 1 above)
Comment (ghost) [reply src/CFBG.cpp:334]: It is needed - it's a crucial change of this PR.

**Comment by ghost** [comment]
@Winfidonarleyan Your BG queue rewrite supersedes this one?

**Comment by FrancescoBorzi** [comment]
@UltraNix I wouldn't wait for that rewrite, I'd merge this meanwhile if it is ready to go

**Comment by ghost** [comment]
@FrancescoBorzi I tested that and no regression seen.

**Pull request merged**